pkg/batcher/createfleet.go

```diff
@@ -52,57 +52,82 @@ func (b *CreateFleetBatcher) CreateFleet(ctx context.Context, createFleetInput *
 	return result.Output, result.Err
 }
 
+// processInstances processes the instances from a CreateFleet output and returns the new request index
+func processInstances(ctx context.Context, output *ec2.CreateFleetOutput, totalCapacity int, results []Result[ec2.CreateFleetOutput], startIdx int) ([]Result[ec2.CreateFleetOutput], int) {
```

Comment:
Is this change proposing calling CreateFleet multiple times? The SDK does this by default for retryable errors, so I'm curious if we need to add our own retry logic on top of what the SDK is doing

Reply:
This PR isn't fully ready but ya the SDK does retry however we notice that somethings part of the CreateFleet fails (partially fulfillment) and the rest of the instances within the batch fails. I am not sure the SDK would retry these failures. I wanted to make the batcher retry the instances that came back fail because the time to recreate nodeclaim is pretty slow

Reply:
Ah i think the instance type got iced by rate limit. I will do some digging to make sure that is the case but @jonathan-innis does it make sense to avoid icing any instance types if the error is request limit related? Do we classify errors that are ICE-able vs those we ignore?